src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/create/actions/CompleteFlowCreateAction.java

```diff
@@ -87,6 +93,33 @@ protected void perform(State from, State to, Event event, FlowCreateContext cont
         });
 
         dashboardLogger.onFlowStatusUpdate(flowId, flowStatus);
-        stateMachine.saveActionToHistory(format("The flow status was set to %s", flowStatus));
+        saveHistoryWithDump(stateMachine, flowStatus);
+    }
+
+    private void saveHistoryWithDump(FlowCreateFsm stateMachine, FlowStatus flowStatus) {
+        Optional<Flow> flow = flowRepository.findById(stateMachine.getFlowId());
+        Optional<FlowPath> forwardPath = flowPathRepository.findById(stateMachine.getForwardPathId());
+        Optional<FlowPath> reversePath = flowPathRepository.findById(stateMachine.getReversePathId());
+        Optional<FlowPath> protectedForwardPath = flowPathRepository.findById(stateMachine.getProtectedForwardPathId());
+        Optional<FlowPath> protectedReversePath = flowPathRepository.findById(stateMachine.getProtectedReversePathId());
+
+        if (flow.isPresent() && forwardPath.isPresent() && reversePath.isPresent()) {
+            stateMachine.saveActionWithDumpToHistory(format("The flow status was set to %s", flowStatus),
```

Comment:
This action message must be exactly like it is now, otherwise flow status will not be displayed in flow history API (despite the history is saved ok).